live-data-collector/src/main/scala/pl/touk/nussknacker/engine/livedata/LiveDataUploaderHolder.scala

```diff
@@ -0,0 +1,87 @@
+package pl.touk.nussknacker.engine.livedata
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import org.slf4j.LoggerFactory
+import pl.touk.nussknacker.engine.api.process.ProcessIdWithName
+import pl.touk.nussknacker.engine.livedata.LiveDataUploader.LiveDataUploaderConfig
+import pl.touk.nussknacker.engine.newdeployment.DeploymentId
+
+import java.time.Instant
+import java.util.concurrent.{Executors, ScheduledExecutorService, ScheduledFuture, TimeUnit}
+import scala.compat.java8.FunctionConverters.asJavaFunction
+
+private[livedata] object LiveDataUploaderHolder {
+
+  private val activeUploaders =
+    Caffeine
+      .newBuilder()
+      .expireAfterAccess(java.time.Duration.ofHours(1))
+      .build[
+        CacheKey,
+        String
+      ]() // Should be [(ProcessIdWithName, DeploymentId), Unit], but Scala 2.12 does not accept tuples and Unit
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def getExistingOrStartLiveDataUploader(
+      processIdWithName: ProcessIdWithName,
+      deploymentIdOpt: Option[DeploymentId],
+      config: LiveDataUploaderConfig,
+  ): Unit = {
+    deploymentIdOpt match {
+      case Some(deploymentId) =>
+        activeUploaders.get(
+          CacheKey(processIdWithName, deploymentId),
+          asJavaFunction((cacheKey: CacheKey) => {
+            startPeriodicLiveDataUploader(cacheKey.processIdWithName, cacheKey.deploymentId, config)
+            ""
+          })
+        )
+      case None =>
+        logger.error(
+          "Live data uploader cannot be started, because the DeploymentId is not defined or has non-UUID underlying value"
+        )
+    }
+  }
+
+  def startPeriodicLiveDataUploader(
```

Comment:
- when some event is captured by LiveDataCollecting listener (something happens in the scenario and it is recorded by LiveData mechanism), then it is checked whether or not the db uploading is already started and working
- if in that situation db uploading is not working, then it is started, as before with configurable interval, but using `ScheduledExecutorService`
- inactivity (lack of events collected by LiveDataCollectingListener) causes the db uploading to stop (with configurable inactivity timeout)